app/src/main/java/ingsw/group1/findmyphone/log/items/LogItemFormatter.java

```diff
@@ -0,0 +1,192 @@
+        resources = context.getResources();
+        cachedDrawables = EventType.getCachedDrawables(context);
+    }
+
+    /**
+     * Method to format an SMSLogEvent. Name and Address are kept the same. Time is formatted as
+     * a Date. Extra is formatted as specified in {@link LogItemFormatter#formatExtra(SMSLogEvent)}.
+     *
+     * @param eventToFormat The event to format. Unknown type events can not be parsed through
+     *                      this method.
+     * @return The formatted {@link LogItem}. Returns {@code null} if the event is of an unknown
+     * type or if the drawable icon could not be loaded.
+     */
+    @Nullable
+    public LogItem formatItem(SMSLogEvent eventToFormat) {
+        if (eventToFormat.getType() == EventType.UNKNOWN)
+            return null;
+        Drawable appropriateDrawable = cachedDrawables.get(eventToFormat.getType());
+        if (appropriateDrawable == null)
+            return null;
+
+        String
+                formattedName = formatName(eventToFormat),
+                formattedAddress = formatAddress(eventToFormat),
+                formattedTime = formatDate(eventToFormat),
+                formattedExtra = formatExtra(eventToFormat);
+
+        GeoPosition positionIfAny = formatPosition(eventToFormat);
+
+        return new LogItem(
+                formattedAddress,
+                formattedName,
+                formattedTime,
+                formattedExtra,
+                appropriateDrawable,
+                eventToFormat.getTime(),
+                positionIfAny
+        );
+    }
+
+    /**
+     * Method used to format a collection of events. Items that return {@code null} when
+     * formatted are excluded from the resulting list.
+     *
+     * @param eventsToFormat A Collection of events to format.
+     * @return A {@link List} of {@link LogItem}, ready to be displayed.
+     */
+    @NonNull
+    public List<LogItem> formatItems(@NonNull Collection<SMSLogEvent> eventsToFormat) {
+        List<LogItem> result = new ArrayList<>();
+        for (SMSLogEvent event : eventsToFormat) {
+            LogItem formattedItem = formatItem(event);
+            if (formattedItem != null) result.add(formattedItem);
+        }
+        return result;
+    }
+
+    /**
+     * Method to get an appropriate name for the contact associated to the event. The name is
+     * looked up in the {@link SMSContactManager} class, if no matching contact is found a
+     * default value (an empty String) is returned.
+     *
+     * @param event The event for which the name must be formatted.
+     * @return A String containing the name for the Contact if it was present in the contacts
+     * database, or an empty String otherwise.
+     */
+    @NonNull
+    private String formatName(@NonNull SMSLogEvent event) {
+        SMSContact contact = contacts.getContactForAddress(event.getAddress());
+        if (contact != null) return contact.getName();
```

Comment:
In case you make the change i suggested you on the constructor of smsLogEvent you can also remove this if statement and also the method getContactForAddress will become NonNull

Reply:
Well even if an address is valid it's not guaranteed to be contained in the contacts database, so null might always be returned by the `SMSContactManager`